core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -77,7 +77,12 @@ object StaticFile {
         .flatMap(fromURL(_, req)))
   }
 
-  def fromURL[F[_]](url: URL, req: Option[Request[F]] = None)(implicit
+  def fromURL[F[_]: Sync: Files](
+      url: URL,
+      req: Option[Request[F]] = None): OptionT[F, Response[F]] =
+    fromURL(url, req, calcETag[F])
+
+  def fromURL[F[_]](url: URL, req: Option[Request[F]], etagCalculator: File => F[String])(implicit
       F: Sync[F]): OptionT[F, Response[F]] = {
     val fileUrl = url.getFile()
```

Comment:
This line seems buggy to me:

> Gets the file name of this URL. The returned file portion will be the same as getPath(), plus the concatenation of the value of getQuery(), if any. If there is no query portion, this method and getPath() will return identical results.

If the URL is `https://www.example.com/blah`, the result will be `/blah`, and looked up on the local file system.  I think we can only use `file` if the URL scheme is `file://`.

Reply:
that means that it also a security issue.